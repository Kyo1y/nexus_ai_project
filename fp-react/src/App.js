import React from 'react'
import {HashRouter} from 'react-router-dom'
import Router from 'Router'
import AuthProvider from 'context/auth.provider'
import {AppProvider} from './context/app.context'
import {ChatProvider} from './context/chat.context'
import {PromptProvider} from './context/prompt.context'
import {hasSession, getSession} from 'utils/session'
import {ToastContainer} from 'react-toastify'
import Header from 'components/Header/Header'
import Footer from 'components/Footer/Footer'
import 'react-toastify/dist/ReactToastify.min.css'
import 'utils/icons'
import 'react-perfect-scrollbar/dist/css/styles.css'

const App = () => {
  let session
  let fullName, id, email, roles

  if (hasSession()) {
    session = getSession()
    fullName = session && session.name
    id = session && session.loggedInAs
    email = session && session.email
    roles = session && session.roles
  }

  return (
    <AppProvider>
      <PromptProvider>
        <ChatProvider>
          <AuthProvider
            value={{
              id,
              fullName,
              email,
              roles,
            }}
          >
            <HashRouter>
              <Header/>
              <Router/>
              <Footer/>
              <ToastContainer/>
            </HashRouter>
          </AuthProvider>
        </ChatProvider>
      </PromptProvider>
    </AppProvider>
  )
}

export default App
