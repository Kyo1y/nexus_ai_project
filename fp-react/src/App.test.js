import React from 'react'
import {shallowToJson} from 'enzyme-to-json'
import {shallow} from 'enzyme'
import App from './App'

it('renders without crashing', () => {
  shallow(<App />)
})

it('matches the snapshot', () => {
  const component = shallow(<App />)

  expect(shallowToJson(component)).toMatchSnapshot()
})
