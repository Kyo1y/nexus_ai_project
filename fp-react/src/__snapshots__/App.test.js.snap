// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<AppProvider>
  <div
    className="app"
  >
    <HashRouter>
      <Header />
      <Router />
      <Footer />
      <withRouter(ScrollToTop) />
      <ToastContainer
        autoClose={5000}
        bodyClassName={null}
        className={null}
        closeButton={
          <CloseButton
            ariaLabel="close"
          />
        }
        closeOnClick={true}
        draggable={true}
        draggablePercent={80}
        hideProgressBar={false}
        newestOnTop={false}
        pauseOnFocusLoss={true}
        pauseOnHover={true}
        position="top-right"
        progressClassName={null}
        progressStyle={null}
        role="alert"
        rtl={false}
        style={null}
        toastClassName={null}
        transition={[Function]}
      />
    </HashRouter>
  </div>
</AppProvider>
`;
